### PR TITLE
Use range-based for loops

### PR DESCRIPTION
**Pull request by Broxzier**
This PR tries to clean up the code a little bit by using range-based for loops instead of standard for loops, where the index was only being used for random access in the array it looped over. It uses `auto` for types, and `auto &` for instances.

### CODE REVIEW

File: src/openrct2/interface/Theme.cpp [hunk 1]
```diff
@@ -337,12 +337,11 @@ void UITheme::SetName(const utf8 * name)
 
 const UIThemeWindowEntry * UITheme::GetEntry(rct_windowclass windowClass) const
 {
-    for (size_t i = 0; i < Entries.size(); i++)
+    for (auto & entry : Entries)
```
Comment (IntelOrca) [inline src/openrct2/interface/Theme.cpp]: This can be `const`

File: src/openrct2/localisation/LanguagePack.cpp [hunk 2]
```diff
@@ -621,9 +619,9 @@ class LanguagePack final : public ILanguagePack
         if (_currentGroup == nullptr)
         {
             // Make sure the list is big enough to contain this string id
-            while (_strings.size() <= (size_t)stringId)
+            if ((size_t)stringId >= _strings.size())
             {
-                _strings.push_back(nullptr);
+                _strings.resize(stringId + 1);
```
Comment (IntelOrca) [inline src/openrct2/localisation/LanguagePack.cpp]: Does `resize` guarantee `nullptr`?

File: src/openrct2/scenario/ScenarioRepository.cpp [hunk 3]
```diff
@@ -367,28 +367,26 @@ class ScenarioRepository final : public IScenarioRepository
...
+                return &scenario;
             }
         }
         return nullptr;
     }
 
     const scenario_index_entry * GetByPath(const utf8 * path) const override
     {
-        for (size_t i = 0; i < _scenarios.size(); i++)
+        for (auto & scenario : _scenarios)
```
Comment (IntelOrca) [inline src/openrct2/scenario/ScenarioRepository.cpp]: `const`

File: src/openrct2/scenario/ScenarioRepository.cpp [hunk 4]
```diff
@@ -367,28 +367,26 @@ class ScenarioRepository final : public IScenarioRepository
 
     const scenario_index_entry * GetByFilename(const utf8 * filename) const override
     {
-        for (size_t i = 0; i < _scenarios.size(); i++)
+        for (auto & scenario : _scenarios)
```
Comment (IntelOrca) [inline src/openrct2/scenario/ScenarioRepository.cpp]: `const`

File: src/openrct2/localisation/LanguagePack.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/localisation/LanguagePack.cpp]: It creates a default value, so in this case it is `nullptr`, yes.

**Comment by janisozaur** [comment]
Our tests caught something! Amazing!

File: src/openrct2/audio/Audio.cpp [hunk 5]
```diff
@@ -160,11 +160,11 @@ void audio_populate_devices()
     std::vector<std::string> devices = audioContext->GetOutputDevices();
 
     // Replace blanks with localised unknown string
-    for (size_t i = 0; i < devices.size(); i++)
+    for (auto & device : devices)
```
Comment (Gymnasiast) [inline src/openrct2/audio/Audio.cpp]: This is very nitpicky of me, but we use `auto &device` throughout.

File: src/openrct2/localisation/LanguagePack.cpp (hunk 2 above)
Comment (Broxzier) [reply src/openrct2/localisation/LanguagePack.cpp]: `resize` defaults initializes, `reserve` does not.

File: src/openrct2/network/NetworkGroup.cpp [hunk 6]
```diff
@@ -44,10 +44,7 @@ NetworkGroup NetworkGroup::FromJson(const json_t * json)
 
     group.Id    = (uint8)json_integer_value(jsonId);
     group._name = std::string(json_string_value(jsonName));
-    for (auto &action : group.ActionsAllowed)
-    {
-        action = 0;
-    }
+    std::fill(group.ActionsAllowed.begin(), group.ActionsAllowed.end(), 0);
```
Comment (Broxzier) [inline src/openrct2/network/NetworkGroup.cpp]: I don't think this is even needed. When `group` is initialized, it already sets everything to their default.

**Comment by ZehMatt** [comment]
To add my 5 cents, you should perhaps use const in some cases.

**Comment by Broxzier** [comment]
@ZehMatt `const` is already being used when the element does not get changed.

**Comment by Gymnasiast** [comment]
@IntelOrca Is this good to merge?

**Pull request merged**